reconcile/gitlab_housekeeping.py

```diff
@@ -209,13 +212,54 @@ def merge_merge_requests(dry_run, gl, merge_limit, rebase, insist=False,
                     logging.error('unable to merge {}: {}'.format(mr.iid, e))
 
 
+# Each '/lgtm' must store the changes (diff) it is approving
+def store_diff_when_lgtm(dry_run, gl, accounts, settings):
```

Comment:
> i think that maybe you should approach this from gitlab-owners

From what I can understand, `gitlab-owners`,  has the primary responsibility of adding the approval label on MRs. It should use the LGTM -> diff mapping (on S3) to decide if an MR should have an approval label. Creating this mapping itself shouldn't be `gitlab-owner`'s responsibility IMO. 

How about creating a separate integration for mapping LGTMs to thier corresponding diffs?

Reply:
gitlab-owners works in the context of a MR. it can be the one to check if there was no new code (or, rebase) after the last lgtm label. this may help in not keeping anything in a state at all.

my proposal is - only add the label when all conditions are true. this should also be opt-in of course through the app-interface schema.